### PR TITLE
Improve printing of expected patterns in parse errors.

### PR DESCRIPTION
**Pull request by eddyb**
Now they should look identical to syntax that can be used to create them.
*Blocked on https://github.com/LykenSol/grammer/pull/7 and https://github.com/rust-lang/gll/pull/123.*

### CODE REVIEW

File: macros/tests/json.rs [hunk 1]
```diff
@@ -15,9 +15,39 @@ mod json_like {
...
+    };
+
+    // HACK(eddyb) clean up the result, as we have no span info.
+    let result = result
+        .replace("Span", "?")
+        .replace("?..? => ", "")
+        .replace("?..?", "?");
+
+    // FIXME(eddyb) Remove this trailing-comma-ignoring hack
+    // once rust-lang/rust#59076 reaches the stable channel.
```
Comment (CAD97) [inline macros/tests/json.rs]: Isn't this stable already?

File: macros/tests/json.rs [hunk 2]
```diff
@@ -15,9 +15,39 @@ mod json_like {
...
+
+    let result = json_like::Value::parse(tokens);
+    let result = match &result {
+        Ok(result) => format!("{:#?}", result),
+        Err(gll::grammer::parser::ParseError { at, expected }) => {
+            format!("{:?}: error: expected {:?}", at, expected)
+        }
+    };
+
+    // HACK(eddyb) clean up the result, as we have no span info.
```
Comment (CAD97) [inline macros/tests/json.rs]: We should just bite the bullet and run tests with `--cfg procmacro2_semver_exempt`.

Comment (eddyb) [reply macros/tests/json.rs]: What I don't get is why don't they always have the `fmt::Debug` implementation enabled.

Comment (eddyb) [reply macros/tests/json.rs]: Also this doesn't work on stable Rust FWIW.

**Pull request merged**